lib/iris/coords.py

```diff
@@ -1684,92 +1745,6 @@ def xml_element(self, doc):
-
-        return new_coord
-
-    def _sanitise_array(self, src, ndmin):
-        if is_lazy_data(src):
-            # Lazy data : just ensure ndmin requirement.
-            ndims_missing = ndmin - src.ndim
-            if ndims_missing <= 0:
-                result = src
-            else:
-                extended_shape = tuple([1] * ndims_missing + list(src.shape))
-                result = src.reshape(extended_shape)
-        else:
-            # Real data : a few more things to do in this case.
-            # Ensure the array is writeable.
-            # NB. Returns the *same object* if src is already writeable.
-            result = np.require(src, requirements='W')
-            # Ensure the array has enough dimensions.
-            # NB. Returns the *same object* if result.ndim >= ndmin
-            result = np.array(result, ndmin=ndmin, copy=False)
-            # We don't need to copy the data, but we do need to have our
-            # own view so we can control the shape, etc.
-            result = result.view()
-        return result
-
-    @property
-    def points(self):
-        """The local points values as a read-only NumPy array."""
-        return self._points_dm.data
-
-    @points.setter
-    def points(self, points):
-        # Set the points to a new array - as long as it's the same shape.
-
-        # Ensure points has an ndmin of 1 and is either a numpy or lazy array.
-        # This will avoid Scalar coords with points of shape () rather
-        # than the desired (1,).
-        points = self._sanitise_array(points, 1)
-
-        # Set or update DataManager.
-        if self._points_dm is None:
-            self._points_dm = DataManager(points)
-        else:
-            self._points_dm.data = points
-
-    @property
-    def bounds(self):
-        """
-        Property containing the bound values, as a NumPy array,
-        or None if no bound values are defined.
-
-        .. note:: The shape of the bound array should be: ``points.shape +
-            (n_bounds, )``.
-
-        """
-        bounds = None
-        if self.has_bounds():
-            bounds = self._bounds_dm.data
-        return bounds
-
-    @bounds.setter
-    def bounds(self, bounds):
-        # Ensure the bounds are a compatible shape.
-        if bounds is None:
-            self._bounds_dm = None
-        else:
-            bounds = self._sanitise_array(bounds, 2)
-            if self.shape != bounds.shape[:-1]:
-                raise ValueError("Bounds shape must be compatible with points "
-                                 "shape.")
-            if not self.has_bounds() \
-                    or self.core_bounds().shape != bounds.shape:
-                # Construct a new bounds DataManager.
-                self._bounds_dm = DataManager(bounds)
-            else:
-                self._bounds_dm.data = bounds
-
     # This is necessary for merging, but probably shouldn't be used otherwise.
     # See #962 and #1772.
     def __hash__(self):
```

Comment:
I looked into this.
The same code is present in `Cube.__hash__` (and a couple of others).

***In Python3***, the hash operation is actually disabled for user classes that define `__eq__`, which means you must provide `__hash__` to allow, for example, collecting the objects into sets, or using them as dictionary keys.
So it seems we ***do*** need this as otherwise we can't collect Coords into sets etc.

**The problem:**
This violates a general principle that
"objects which compare equal have the same hash value".
See: https://docs.python.org/2.7/reference/datamodel.html#object.__hash__

So, we should really fix these, 
but it probably means changing other areas such as merge, and possibly concatenate.